### PR TITLE
Added hook_page_attachments() event.

### PR DESCRIPTION
**Pull request by Ambient-Impact**
Hey, I just put this together and thought others might need it too. Feel free to edit as needed. I don't have a local testing set up, so I haven't tested the test (how meta).

### CODE REVIEW

File: tests/src/Unit/Page/PageAttachmentsEventTest.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+use Drupal\Tests\UnitTestCase;
+
+/**
+ * Class PageAttachmentsTest.
+ *
+ * @package Drupal\Tests\hook_event_dispatcher\Unit\Page
+ *
+ * @group hook_event_dispatcher
+ */
+class PageAttachmentsTest extends UnitTestCase {
```
Comment (pdenooijer) [inline tests/src/Unit/Page/PageAttachmentsEventTest.php]: Please rename this to `PageAttachmentsEventTest` as the name of the file and the `PageAttachmentsEvent`

File: tests/src/Unit/Page/PageAttachmentsEventTest.php [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  /**
+   * Test the PageAttachments by reference.
+   */
+  public function testPageAttachmentsByReference() {
+    $testAttachments = ['#attached' => ['library' => ['test/test']]];
+    $expectedBuild['new'] = $testAttachments;
+
+    $this->manager->setEventCallbacks([
+      HookEventDispatcherInterface::PAGE_ATTACHMENTS => function (PageAttachments $event) use ($testAttachments) {
```
Comment (pdenooijer) [inline tests/src/Unit/Page/PageAttachmentsEventTest.php]: This should be `PageAttachmentsEvent`.

File: tests/src/Unit/Page/PageAttachmentsEventTest.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+  /**
+   * Test the PageAttachments by set.
+   */
+  public function testPageAttachmentsBySet() {
+    $testAttachments = ['#attached' => ['library' => ['test/test']]];
+    $expectedBuild['new'] = $testAttachments;
+
+    $this->manager->setEventCallbacks([
+      HookEventDispatcherInterface::PAGE_ATTACHMENTS => function (PageAttachments $event) use ($testAttachments) {
```
Comment (pdenooijer) [inline tests/src/Unit/Page/PageAttachmentsEventTest.php]: This should be `PageAttachmentsEvent`.

**Review by pdenooijer — CHANGES_REQUESTED** [review]
Thanks for the contribution, seems nice! Just some small fixes then I will merge this 👍

**Comment by Ambient-Impact** [comment]
I'm new to changing a pull request and the GitHub documentation isn't very helpful; should I make these changes as a new commit, or is there a better way?

File: tests/src/Unit/Page/PageAttachmentsEventTest.php (hunk 1 above)
Comment (Ambient-Impact) [reply tests/src/Unit/Page/PageAttachmentsEventTest.php]: To clarify: the name of the class should be ```PageAttachmentsEventTest``` to match the file name?

Comment (pdenooijer) [reply tests/src/Unit/Page/PageAttachmentsEventTest.php]: Yes

**Comment by pdenooijer** [comment]
Yes just amend the commit and force push this, so we have a clean history.

File: tests/src/Unit/Page/PageAttachmentsEventTest.php (hunk 1 above)
Comment (Ambient-Impact) [reply tests/src/Unit/Page/PageAttachmentsEventTest.php]: Ah yeah, that makes sense. Programming while tired seems to lead to these silly mistakes. :P

**Comment by Ambient-Impact** [comment]
Alright, let me know if that's okay or if it needs some more edits.

File: tests/src/Unit/Page/PageAttachmentsEventTest.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Drupal\Tests\hook_event_dispatcher\Unit\Page;
+
+use Drupal\Core\DependencyInjection\ContainerBuilder;
+use Drupal\hook_event_dispatcher\Event\Page\PageAttachments;
```
Comment (pdenooijer) [inline tests/src/Unit/Page/PageAttachmentsEventTest.php]: Please update this use statement as well, so the Event can be used 😉

**Review by pdenooijer — CHANGES_REQUESTED** [review]
Please fix the test

File: tests/src/Unit/Page/PageAttachmentsEventTest.php (hunk 4 above)
Comment (Ambient-Impact) [reply tests/src/Unit/Page/PageAttachmentsEventTest.php]: 🤦‍♀ Can't believe I missed that. Thanks for your patience.

**Review by pdenooijer — APPROVED** [review]
Thanks, I'll fix the test myself 👍

**Comment by Ambient-Impact** [comment]
I appreciate it. I'll have to get local testing set up before opening any other pull requests.

**Comment by pdenooijer** [comment]
`composer install` && `composer test` should do the trick in the module 🙂

**Pull request merged**